File: public/actions/python/decryptDataLocally.py

from cloudproof_py.cover_crypt import CoverCrypt


def decrypt_data_locally(user_key: string, ciphertext: string):
    """Decrypt data locally"""
    cover_crypt = CoverCrypt()
    plaintext = cover_crypt.decrypt(user_key,ciphertext)
    return plaintext

File: public/actions/javascript/decryptDataLocally.ts
import { CoverCrypt, PrivateKey } from "cloudproof_js";

//
// Decrypt data locally
//
export const decryptDataLocally = async (encryptedData: Uint8Array, decryptionKey: Uint8Array | PrivateKey): Promise<string> => {
  const { CoverCryptHybridDecryption } = await CoverCrypt();
  const decrypter = new CoverCryptHybridDecryption(decryptionKey);
  const decryptedData = decrypter.decrypt(encryptedData);
  return new TextDecoder().decode(decryptedData.plaintext);
};
